### PR TITLE
refactor: 💡 Convert SQFormCheckboxGroupItem to Typescript

### PR DESCRIPTION
**Pull request by abbas-ali123**
Loom of functionality in storybook
https://www.loom.com/share/6196c918fcba4802b48729cf341d813c

✅ Closes: #335

### CODE REVIEW

**Review by SeanGroff — APPROVED** [review]

**Review by caleboleary — APPROVED** [review]

File: src/components/SQForm/SQFormCheckboxGroupItem.tsx [hunk 1]
```diff
@@ -14,6 +13,23 @@ const useStyles = makeStyles(theme => ({
...
 
+interface SQFormCheckboxGroupItemProps {
+  /** The name of the group this checkbox is a part of */
+  groupName: string;
+  /** Label for the checkbox */
+  label: string;
+  /** Value for the checkbox */
+  value: string | boolean | number;
+  /** Function to call when input value is changed */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (codyarose) [inline src/components/SQForm/SQFormCheckboxGroupItem.tsx:24]: We can use `CheckboxProps` here as well, to get the `onChange` type that `<Checkbox />` expects: 
```ts
onChange: CheckboxProps['onChange'];
````

**Review by codyarose — CHANGES_REQUESTED** [review]

**Comment by codyarose** [comment]
When committing changes via `npm run commit` you'll want to just press `enter` to move passed the `BREAKING CHANGES` question if there are no breaking changes. Anything other than an empty value will trigger a major version update by semver.
You may be able to do something like [this](https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message#amending-older-or-multiple-commit-messages) to remove that breaking change section of the commit message.

File: src/components/SQForm/SQFormCheckboxGroupItem.tsx (hunk 1 above)
Comment (abbas-ali123) [reply src/components/SQForm/SQFormCheckboxGroupItem.tsx:24]: @codyarose The useForm component does not support that. I have tried and it gives me this error 
image

Comment (codyarose) [reply src/components/SQForm/SQFormCheckboxGroupItem.tsx:24]: Ah, right. My mistake, that should be possible after I add the useForm generics.

**Review by codyarose — APPROVED** [review]

**Pull request merged**